### PR TITLE
[luci] Introduce MigrateLegacyShapeDtypePass in luci

### PR DESCRIPTION
**Pull request by llFreetimell**
Parent Issue : #4796

Until refactoring for removing dependency between `loco` shape and circle shape is finished,
shape and dtype of `loco` should be copied to circle nodes to keep consistency.
This commit will introduce `CopyLocoItemsToCirclePass` and this pass will do it.
FYI, be careful that when copying `loco` shape to circle node, shape signature of the circle node
is considered, but shape of `loco` is not changed.

When the refactoring is finished, this pass will be removed.

ONE-DCO-1.0-Signed-off-by: Seok NamKoong <seok9311@naver.com>

### CODE REVIEW

**Comment by seanshpark** [comment]
Pass name `CopyLocoItemsToCircle` is somewhat strange...

**Comment by seanshpark** [comment]
How about `MigrateLegacyShapeDTypePass` ?

**Comment by llFreetimell** [comment]
`MigrateLegacyShapeDtypePass` looks elegant, I will use it :)

**Comment by mhs4670go** [comment]
(optional) Umm.. `MigrateCanonicalShapeDtypePass` ..?

File: compiler/luci/pass/src/MigrateLegacyShapeDtypePass.cpp [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+    {
+      auto loco_shape = loco::shape_get(node).as<loco::TensorShape>();
+
+      // When shape of loco is copied to circle node, ShapeSignature should be applied.
+      loco::TensorShape new_shape;
+      new_shape.rank(loco_shape.rank());
+      for (uint32_t i = 0; i < loco_shape.rank(); ++i)
+      {
+        if (circle_node->shape_signature().rank() > 0 &&
+            circle_node->shape_signature().dim(i) == -1)
```
Comment (seanshpark) [inline compiler/luci/pass/src/MigrateLegacyShapeDtypePass.cpp:76]: I'm not sure for this one but can `circle_node->shape_signature().rank() == 2` and `loco_shape.rank() == 3`  ?
or maybe add `assert(circle_node->shape_signature().rank() == loco_shape.rank())` ?

**Comment by seanshpark** [comment]
When this lands and this Pass is on, please run _circle2circle_ with our internal models...

**Comment by seanshpark** [comment]
(optional) it would be nice if you could add comments about what the intention of the codes are...
after some months or years passes, we may forget them...

File: compiler/luci/pass/src/MigrateLegacyShapeDtypePass.cpp (hunk 1 above)
Comment (llFreetimell) [reply compiler/luci/pass/src/MigrateLegacyShapeDtypePass.cpp:76]: Those error will be checked at following codes.
(Following codes are in master branch)

https://github.com/Samsung/ONE/blob/5a2e89734126996dfbe90d2defc2b1474b623c60/compiler/luci/service/src/Validate.cpp#L121-L127

However, adding `assert` looks good! I will add it :)

---

FYI, I am planning to validate refactoring codes more strictly by following codes.
(It is draft code)
https://github.com/Samsung/ONE/blob/42a5ce43f46e3e22f55b212dcc5d631269381154/compiler/luci/pass/src/CheckCircleRulesPass.cpp#L103-L136

**Review by seanshpark — APPROVED** [review]
LGTM thank you!

**Review by mhs4670go — APPROVED** [review]
LGTM

**Pull request merged**